src/Feature/Editor/EditorDiffMarkers.re

```diff
@@ -55,9 +55,13 @@ let generate = buffer =>
          );
 
        // if there are deleted lines past the end of the current document, mark the last lines as having deletes afterwards, or being modified
-       if (Array.length(deletes) - shift^ - 1 > Array.length(adds)) {
-         markers[Array.length(markers) - 1] = (
-           switch (markers[Array.length(markers) - 1]) {
+       let markerLen = Array.length(markers);
+       if (markerLen > 0
```

Comment:
I think this is safe, but just wanted to double-check!

`refmt` sure made it hard to parse that logic ☹️

Reply:
Seems safe, but not correct. If you've deleted the entire document there's definitely deletes past the end, so there should be a marker for that. So perhaps just special case it, `if (markers == [||]) { [|DeletedBefore|] } else { ... }`.

Also, `markerLen` sounds like it describes the length of a single marker, not the array of markers. I'm not sure I see the point of adding this indirection anyway though, since `Array.length` doesn't computer anything the cost should be the same.

Reply:
Oh, good catch, sorry I missed that case. Implemented the suggestion.

Reply:
Looks like `refmt` is happier this time.